### PR TITLE
Secboot implementation

### PR DESCRIPTION
**Pull request by covanam**

### CODE REVIEW

File: test/tinycrypt/lib/source/ecc_platform_specific.c [hunk 1]
```diff
@@ -77,29 +72,10 @@ int default_CSPRNG(uint8_t *dest, unsigned int size) {
...
 
-  int fd = open("/dev/urandom", O_RDONLY | O_CLOEXEC);
-  if (fd == -1) {
-    fd = open("/dev/random", O_RDONLY | O_CLOEXEC);
-    if (fd == -1) {
-      return 0;
-    }
-  }
+  /* dummy random for test purpose */
+  while (size--)
```
Comment (nicolasmoro) [inline test/tinycrypt/lib/source/ecc_platform_specific.c]: You can also use the libc rand, which you can seed with a constant value

File: test/tinycrypt/tests/secboot_hmac.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	struct tc_hmac_state_struct state;
+	uint8_t digest[32];
+
+	tc_hmac_set_key(&state, secret_key, sizeof(secret_key));
+
+	tc_hmac_init(&state);
+	tc_hmac_update(&state, firmware, sizeof(firmware));
+	tc_hmac_final(digest, 32, &state);
+
+	if (compare(digest, tag, 32)) {
```
Comment (nicolasmoro) [inline test/tinycrypt/tests/secboot_hmac.c]: A point of detail, but a digest is the output of a hashing function. Here you are comparing two HMAC tags.

File: test/tinycrypt/tests/secboot_sha256.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+int main() {
+	struct tc_sha256_state_struct state;
+	uint8_t digest[32];
+
+	tc_sha256_init(&state);
+	tc_sha256_update(&state, firmware, sizeof(firmware) - 1);
+	tc_sha256_final(digest, &state);
+
+	if (compare(digest, hash, 32)) {
```
Comment (nicolasmoro) [inline test/tinycrypt/tests/secboot_sha256.c]: Digest and hash mean the same thing, I would call them something like reference_digest and computed_digest, or reference_hash and computed_hash

File: test/tinycrypt/tests/secboot_hmac.c (hunk 2 above)
Comment (covanam) [reply test/tinycrypt/tests/secboot_hmac.c]: renamed to `reference_tag` and `computed_tag`

File: test/tinycrypt/tests/secboot_sha256.c (hunk 3 above)
Comment (covanam) [reply test/tinycrypt/tests/secboot_sha256.c]: changed as suggested

File: test/tinycrypt/lib/source/ecc_platform_specific.c (hunk 1 above)
Comment (covanam) [reply test/tinycrypt/lib/source/ecc_platform_specific.c]: This approach would take more work than what it's worth, as I currently do not link with standard C library but using my own implementation of `printf` and `malloc` ...

I think for our purpose, this should do. But if you insist on having an actual random function, then I do have a random function that I can use here.

Comment (covanam) [reply test/tinycrypt/lib/source/ecc_platform_specific.c]: actually while deleting some code (to resolve a flash overflow problem), I discovered this function is not needed at all. So I deleted it.

**Pull request merged**